### PR TITLE
888711-Prepare the UG documentation for the selection controller

### PR DESCRIPTION
**Pull request by ShaliniSF79**
- Added the Selection Controller behavior

### CODE REVIEW

File: MAUI/DataGrid/selection.md [hunk 1]
```diff
@@ -556,4 +556,108 @@ By default, the keyboard navigation will be enabled when setting the selection a
                     ItemsSource="{Binding OrderInfoCollection}" >
 </syncfusion:SfDataGrid>
 {% endhighlight %}
+{% endtabs %}
+
+
+## Customizing the selection controller behavior
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/selection.md:562]: Customize selection behavior

File: MAUI/DataGrid/selection.md [hunk 2]
```diff
@@ -556,4 +556,108 @@ By default, the keyboard navigation will be enabled when setting the selection a
...
+
+For customizing the selection controller behavior, initially create the instance for selection controller then, we have to create a class by inheriting the `DataGridRowSelectionController`. And then override the the methods you want to customize.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/selection.md:564]: The default keyboard selection behaviors can be customized by setting the instance of the custom `DataGridRowSelectionController` to the `SfDataGrid.SelectionController` property.

File: MAUI/DataGrid/selection.md [hunk 3]
```diff
@@ -556,4 +556,108 @@ By default, the keyboard navigation will be enabled when setting the selection a
...
+
+### Changing Enter key behavior
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/selection.md:566]: Change enter key behavior

File: MAUI/DataGrid/selection.md [hunk 4]
```diff
@@ -556,4 +556,108 @@ By default, the keyboard navigation will be enabled when setting the selection a
...
+
+By default, pressing the <kbd>Enter</kbd> key moves the current cell focus to the next cell in the same column. To change this behavior, you can override the relevant selection controllers based on the NavigationModes.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/selection.md:568]: By default, while pressing `Enter` key, the current cell will be moved to the next focused cell in the same column. You can change the behavior by overriding the `ProcessKeyDown` method in the custom selection controller.

File: MAUI/DataGrid/selection.md [hunk 5]
```diff
@@ -556,4 +556,108 @@ By default, the keyboard navigation will be enabled when setting the selection a
...
+
+You can modify the <kbd>Enter</kbd> key behavior by overriding the ProcessKeyDown method in the selection controller. In this method, create a new KeyEventArgs instance that simulates a <kbd>Tab</kbd> key press and process the <kbd>Tab</kbd> key action accordingly.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/selection.md:570]: This is not applicable remove it.

File: MAUI/DataGrid/selection.md [hunk 6]
```diff
@@ -556,4 +556,108 @@ By default, the keyboard navigation will be enabled when setting the selection a
...
+                    base.ProcessKeyDown(args, isCtrlKeyPressed, isShiftKeyPressed);
+                    break;
+            }
+        }
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+Additionally, we included a scenario to disable the <kbd>Enter</kbd> key behavior. When the <kbd>Enter</kbd> key is pressed, it does not move to the next row but remains in the same place.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/selection.md:625]: The following code snippets show how to disable the default enter key behavior in `SfDataGrid` by writing custom selection controller.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

File: MAUI/DataGrid/selection.md (hunk 1 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/selection.md:562]: Changed like this

File: MAUI/DataGrid/selection.md (hunk 6 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/selection.md:625]: Changed

File: MAUI/DataGrid/selection.md (hunk 5 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/selection.md:570]: I removed it

File: MAUI/DataGrid/selection.md (hunk 4 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/selection.md:568]: Updated

File: MAUI/DataGrid/selection.md (hunk 3 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/selection.md:566]: Changed

File: MAUI/DataGrid/selection.md (hunk 2 above)
Comment (ShaliniSF79) [reply MAUI/DataGrid/selection.md:564]: Changed

**Review by ashok-kuvaraja — APPROVED** [review]
The changes are fine.

**Pull request merged**